### PR TITLE
enh(ci): add workflow to check pr status

### PR DESCRIPTION
**Pull request by kduret**
## Description

add workflow to check pr status and limit token usage (1 call / minute)

### CODE REVIEW

File: .github/workflows/check-status.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+              fi
+
+              if [ $(echo $CHECK_SUITES | grep -o -i 'failure' | wc -l) -gt 0 ]; then
+                  echo "some jobs are failed"
+                  exit 1
+              fi
+
+              # only remaining pending job should be check-status itself
+              if [ $(echo $CHECK_SUITES | grep -o -i 'pending' | wc -l) -eq 1 ]; then
+                  echo "all jobs are passed"
```
Comment (bdauria) [inline .github/workflows/check-status.yml:47]:
```suggestion
                  echo "all jobs have passed"
```

File: .github/workflows/check-status.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                  exit 1
+              fi
+
+              if [ $(echo $CHECK_SUITES | wc -w) -eq 1 ]; then
+                  echo "this job is the only triggered one"
+                  exit 0
+              fi
+
+              if [ $(echo $CHECK_SUITES | grep -o -i 'failure' | wc -l) -gt 0 ]; then
+                  echo "some jobs are failed"
```
Comment (bdauria) [inline .github/workflows/check-status.yml:41]:
```suggestion
                  echo "some jobs have failed"
```

File: .github/workflows/check-status.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      - release-*
+
+jobs:
+  check-status:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+
+      - run: |
+          sleep 30s
```
Comment (totogtr) [inline .github/workflows/check-status.yml:24]: just for my knowledge, why is a `sleep` needed ?

**Review by totogtr — APPROVED** [review]

File: .github/workflows/check-status.yml (hunk 3 above)
Comment (kduret) [reply .github/workflows/check-status.yml:24]: this is just a defer for api call
it's a subjective value to limit api call usage

**Review by bdauria — APPROVED** [review]

**Pull request merged**